Release branch 2.14 beta (#6527)
PyCQA/pylint: doc/whatsnew/2.14.rst

Summary -- Release highlights
=============================

> In 2.14, we now support python interpreter above 3.7.2.

Reviewer: ```suggestion
With 2.14 ``pylint`` only supports Python version 3.7.2 and above.
```

We introduced several new checks among which ``duplicate-value`` for sets,

> ``comparison-of-constants`` or check related to lambdas. We removed ``no-init`` and

Reviewer 2: ```suggestion
``comparison-of-constants``, and checks related to lambdas. We removed ``no-init`` and
```

made ``no-self-use`` optional as they were too opinionated. We also added an option

> to generate a toml configuration: ``generate-toml-config``

Reviewer: ```suggestion
to generate a toml configuration: ``--generate-toml-config``
```

> to generate a toml configuration: ``--generate-toml-config``

Reviewer 2: ```suggestion
to generate a toml configuration: ``--generate-toml-config``.
```

We migrated to ``argparse`` from ``optparse`` and refactored the configuration handling

> thanks to Daniël van Noords. On the user side it should change the output of the

Reviewer: ```suggestion
thanks to Daniël van Noord. On the user side it should change the output of the
```

Author: Oops sorry !

``--help`` command, and some inconsistencies and bugs should disappear. The behavior

> between option set in config file and in command line will be more consistent. For us,

Reviewer: ```suggestion
between options set in config file and in command line will be more consistent. For us,
```

> between options set in config file and in command line will be more consistent. For us,

Reviewer 2: ```suggestion
between options set in a config file versus on the command line will be more consistent. For us,
```

it will permit to maintain this part of the code easily in the future and anticipate

> optparse removal in python 3.12.

Reviewer: ```suggestion
``optparse's`` removal in Python 3.12.
```

> ``optparse's`` removal in Python 3.12.

Reviewer 2: ```suggestion
``optparse``'s removal in Python 3.12.
```

> As a result of refactor there are a lot of internal deprecation. If you're a library

Reviewer: ```suggestion
As a result of the refactor there are a lot of internal deprecations. If you're a library
```

maintainer that depends on pylint, please verify that you're ready for pylint 3.0
by activating deprecation warnings.

> We continued the integration of ``pylint-error``. `We still welcome any community effort

Reviewer: ```suggestion
We continued the integration of ``pylint-error`` and are now at 33%!. We still welcome any community effort
```

> help to review, integrate, and add good/bad examples <https://github.com/PyCQA/pylint/issues/5953>`_. This should be doable

Reviewer: ```suggestion
to help review, integrate, and add good/bad examples <https://github.com/PyCQA/pylint/issues/5953>`_. This should be doable
```

> without any pylint or astroid knowledge, so this is the perfect entrypoint if you want

Reviewer: ```suggestion
without any ``pylint`` or ``astroid`` knowledge, so this is the perfect entrypoint if you want
```

> to contribute to pylint without investing any time learning the internals.

Reviewer: ```suggestion
to contribute to ``pylint`` or open source without any experience with out code!
```

> to contribute to ``pylint`` or open source without any experience with out code!

Reviewer 2: ```suggestion
to contribute to ``pylint`` or open source without any experience with our code!
```

* The ``config`` attribute of ``PyLinter`` is now of the ``argparse.Namespace`` type instead of
  ``optparse.Values``.

  Ref #5392

* ``UnsupportedAction`` has been deprecated.

  Ref #5392

* ``OptionsManagerMixIn`` has been deprecated.

  Ref #5392

* ``OptionParser`` has been deprecated.

  Ref #5392

* ``Option`` has been deprecated.

  Ref #5392

* ``OptionsProviderMixIn`` has been deprecated.

  Ref #5392

* ``ConfigurationMixIn`` has been deprecated.

* The ``option_groups`` attribute of ``PyLinter`` has been deprecated.

  Ref #5392

* ``get_global_config`` has been deprecated. You can now access all global options from
  ``checker.linter.config``.

  Ref #5392

* ``OptionsManagerMixIn`` has been replaced with ``ArgumentsManager``. ``ArgumentsManager`` is considered
  private API and most methods that were public on ``OptionsManagerMixIn`` have now been deprecated and will
  be removed in a future release.

  Ref #5392

* ``OptionsProviderMixIn`` has been replaced with ``ArgumentsProvider``. ``ArgumentsProvider`` is considered
  private API and most methods that were public on ``OptionsProviderMixIn`` have now been deprecated and will
  be removed in a future release.

>   Ref #5392

Reviewer: This `diff` is horrible, but I'm just going to assume you copied these correctly 😄
